frontend/src/components/organisms/SignupForm.tsx

```diff
@@ -186,6 +194,24 @@ const SignupForm = () => {
             label="Password"
             {...register("password", {
               required: { value: true, message: "Required" },
+              minLength: {
+                value: 6,
+                message: "Password must be at least 6 characters",
+              },
+              validate: {
+                hasUpper: (value) =>
+                  /.*[A-Z].*/.test(value) ||
+                  "Password must contain at least one uppercase letter",
+                hasLower: (value) =>
+                  /.*[a-z].*/.test(value) ||
+                  "Password must contain at least one lowercase letter",
+                hasNumber: (value) =>
+                  /.*[0-9].*/.test(value) ||
+                  "Password must contain at least one number",
+                hasSpecialChar: (value) =>
+                  /.*[\W_].*/.test(value) ||
+                  "Password must contain at least one special character",
+              },
```

Comment:
This might not be necessary 🤔. Firebase handles this already, and it is available with our current implementation.

Reply:
That's interesting because I was totally able to create accounts with password `12345678`. Maybe we just didn't enable it in Firebase yet for testing. Either way, I think there's some value in having `react-hook-form`'s immediate feedback about valid passwords without having to spam signup attempts every single time. This implementation means that the user can correct their password until it's correct without ever having to ping Firebase.

Reply:
You're right, however, I think a password of 12345678 is "correct". It is just a courtesy from us for users to have strong passwords. Fwiw, I couldn't find a list of what it means for a weak password, but there is an error code for [weak password](https://firebase.google.com/docs/auth/flutter/password-auth#create_a_password-based_account), I think a password of under 8 characters is what has triggered it for me in the past!

Reply:
I think it's good practice to enforce stricter password requirements. These are the same requirements that Google requires (see below).

![image](https://github.com/cornellh4i/lagos-volunteers/assets/48730262/32083382-a29d-430a-8666-29d24ac66a49)